### ISSUE TITLE
packaging of sharp binary

### ISSUE THREAD

**Issue by Wintus** [Wintus/lambda-edge-image-resize#5]
original question:

> how did you handle packaging of sharp binary?

_Originally posted by @pontusab in https://github.com/Wintus/lambda-edge-image-resize/issues/4#issuecomment-472909666_

I meant to document the struggle with the binary including shared library (and I didn't so far in commit messages).

In short, I use a webpack option to package node_modules/sharp entirely to keep its path structure.
https://github.com/Wintus/lambda-edge-image-resize/blob/453181393af906afabe25930c333e1f6decab466/lambda/serverless.yml#L15-L17

TODO: more details, existing issues of sharp.

**Comment by pontusab** [comment]
in serverless.yml in webpack config should we add:
```packagerOptions:
      scripts:
        - npm rm sharp && env npm_config_arch=x64 npm_config_platform=linux npm_config_target=8.10.0 npm install sharp```

To install the binary for linux platform?

**Comment by Wintus** [comment]
That script seems to work well.

The official doc suggest the following script:
https://sharp.dimens.io/en/stable/install/#aws-lambda

I cannot check it in non Linux machine since I have Linux only,
but Sharp was deployed without problems when I developed in Ubuntu.

**Comment by Wintus** [comment]
To be solved by layer #7
